src/devices/Arduino/BlockingUnorderedCollection.cs

```diff
@@ -0,0 +1,108 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Iot.Device.Arduino
+{
+    /// <summary>
+    /// Represents a collection that removes objects based on a certain pattern
+    /// </summary>
+    internal class BlockingUnorderedCollection<T>
```

Comment:
Nit: Blocking -> Concurrent?

Reply:
The runtime already has `BlockingCollection<T>` and `ConcurrentBag<T>`, but not the combination I need here. I'll rename to `BlockingConcurrentBag<T>` as this is what this represents.